x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -83,19 +78,29 @@ export const useInitSourcerer = (
     getTimelineSelector(state, TimelineId.active)
   );
 
-  const sourcererScopeSelector = useMemo(() => sourcererSelectors.getSourcererScopeSelector(), []);
-  const {
-    sourcererScope: { selectedDataViewId: scopeDataViewId, selectedPatterns, missingPatterns },
-  } = useDeepEqualSelector((state) => sourcererScopeSelector(state, scopeId));
+  const scopeDataViewId = useSelector((state: State) => {
+    return sourcererSelectors.sourcererScopeSelectedDataViewId(state, scopeId);
+  });
+  const selectedPatterns = useSelector((state: State) => {
+    return sourcererSelectors.sourcererScopeSelectedPatterns(state, scopeId);
+  });
+  const missingPatterns = useSelector((state: State) => {
+    return sourcererSelectors.sourcererScopeMissingPatterns(state, scopeId);
+  });
```

Comment:
This looks great Kevin! Thanks for cleaning and organizing the selectors, this was much needed. I only have one question related to selectors that depend on component props (like `scopeId` here).

The documentation suggests that we should create a local instance of the same selector on each component that uses "parametrized" selectors.  
When reusing this kind of selector for all components, the memorization will be invalidated every time other components that use the same selector render with other values. 

I can imagine a scenario in which the same selector is being used by different components on the same page, with different values, in this scenario the selector would be constantly invalidated. 

I am not sure this could even happen in our app, it's just a hypothetical scenario. And I am also not sure it's worth doing it the other way. I mean, the current changes are already a great improvement, I am just curious about what you think about it.

https://react-redux.js.org/api/hooks#using-memoizing-selectors

Reply:
Yes, that is due to the default cache size of 1, however these selectors all set it to be the maximum number of sourcerer scopes, as of now 4, and so the results are cached. You can test this with the .recomputations() function of a selector, so I added this below:

```
  const testCount = sourcererSelectors.sourcererScopeSelectedDataViewId.recomputations();
  const other1 = useSelector((state: State) => {
    return sourcererSelectors.sourcererScopeSelectedDataViewId(state, scopeId);
  });
  console.log(sourcererSelectors.sourcererScopeSelectedDataViewId.recomputations(), testCount);
```
With the code as it is now, and refreshing the page and nothing else, this logs:
2 2
3 3
3 3
3 3
3 3
4 4
indicating that's how many times it's updated as the page loads. If I remove the 
```
  {
    memoizeOptions: {
      maxSize: SOURCERER_SCOPE_MAX_SIZE,
    },
  }
```
from the selector and log the same value:
3 3
147 147
159 159
235 235
241 241
248 248
304 304
414 414
417 417
432 432
435 435
it shows we are running into exactly this problem, and everything is constantly invalidated. As long as we have a fixed and small number of sourcerer scopes, I think the maxSize solution is fine. The idea of this pr is more to just set things up for moving logic out of useState/useEffects running in a bunch of different components and have it happen in redux instead.

Reply:
There was a page on the reselect docs that went into this part a few months ago, but I can no longer find it ha, sorry. https://reselect.js.org/FAQ#can-i-share-a-selector-across-multiple-component-instances this mention of maxSize is best I can find now.

Reply:
thank you so much for asking the question @semd and for posting the detailed answer @kqualters-elastic I'm learning a lot!

Reply:
@kqualters-elastic Oh that makes sense, I didn't know about the maxSize memorization option. This is a great solution to keep selector usage consistent across components regardless of the number of parameters the selector receives. We just need to keep that in mind when we create the selector.
Thanks!